aten/src/ATen/native/cpu/CatKernel.cpp

```diff
@@ -55,14 +60,24 @@ void cat_serial_kernel_impl(const Tensor& result, const MaterializedITensorListR
   }
 }
 
-void cat_serial_kernel(const Tensor& result, const MaterializedITensorListRef& tensors, int64_t dim) {
-  AT_DISPATCH_FLOATING_TYPES_AND2(kBFloat16, kHalf, result.scalar_type(), "cat_serial_kernel", [&]() {
-    cat_serial_kernel_impl<scalar_t>(result, tensors, dim);
-  });
+void cat_serial_kernel(
+    const Tensor& result,
+    const MaterializedITensorListRef& tensors,
+    int64_t dim) {
+  AT_DISPATCH_FLOATING_TYPES_AND6(
```

Comment:
Use AT_DISPATCH_V2. Makes your life so much easier here `AT_EXPAND(FLOAT8_DTYPES)`

Reply:
This is much nicer, thank you for the tip! The latest commit looks pretty clean.